students/alex_skrn/lesson04/a_new_file.py

```diff
@@ -0,0 +1,180 @@
+"""Assignment: Kata Fourteen: Tom Swift Under Milk Wood."""
+
+import random
+import tkinter as tk
+from tkinter import filedialog
+
+
+# INPUT FILE MANAGEMENT
+def get_filepath():
+    """Return the path to the source file obtained from the user."""
+    # Get the name of the source file from the user.
+    root = tk.Tk()
+    root.withdraw()
+    path = filedialog.askopenfilename()
+    return path
+
+
+def load_file(source_path):
+    """Return a list containing the whole book split on blank space."""
+    # Container for the whole text of the loaded book.
+    book = []
+
+    # Convert the whole book into a list.
+    with open(source_path, 'r') as fromF:
+        for line in fromF:
+            # Take care of paragraphs
+            if line == "\n" or line == "\r\n":
+                line = line.strip()  # Get rid of "\r" or "\n" if any
+                line = line.split()
+                line.append("\n\n")  # Use a double newline instead
+            # Punctuation goes with the words to which it relates
+            else:
+                line = line.strip()
+                line = line.split()
+            book.extend(line)
+
+    return book
+
+
+# OUTPUT FORMAT MANAGEMENT
+def formatter(a_list):
+    """Return a string containing the formatted text."""
+    result = ""
+    for token in a_list:
+        if token == "\n\n":  # Doesn't need a blank space before or after
+            result += token
+        else:
+            result += (token + " ")
+    return result
+
+
+# MAIN PROCESSING
+def get_dict(a_list):
+    """Return a dict for trigram-style text generation."""
+    d = {}
```

Comment:
don't use single character variable names unless its in a for loop and is common nomenclature like:
`for k, v in a_dictionary.items()`  with `k, v` obviously representing `key` and `value`